src/QsCompiler/Transformations/BasicTransformations.cs

```diff
@@ -7,15 +7,33 @@
 using System.Linq;
 using Microsoft.Quantum.QsCompiler.DataTypes;
 using Microsoft.Quantum.QsCompiler.SyntaxTree;
+using Microsoft.Quantum.QsCompiler.Transformations.Core;
 
 
 namespace Microsoft.Quantum.QsCompiler.Transformations.BasicTransformations
 {
     // syntax tree transformations
 
     public class GetSourceFiles :
-        SyntaxTreeTransformation<NoScopeTransformations>
+        QsSyntaxTreeTransformation<GetSourceFiles.TransformationState>
     {
+        public class TransformationState
+        {
+            internal readonly HashSet<NonNullable<string>> SourceFiles = 
+                new HashSet<NonNullable<string>>();
+        }
+
+
+        private GetSourceFiles() :
+            base(new TransformationState())
+        { }
+
+        public override NamespaceTransformation<TransformationState> NewNamespaceTransformation() =>
+            new NamespaceTransformation(this);
```

Comment:
It isn't clear to me what this accomplishes. Doesn't the base NewNamespaceTransformation do the same thing?